src/features/posts/postsSlice.ts

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../../types/Post';
+import { getUserPosts } from '../../api/posts';
+
+type PostsInfo = {
+  items: Post[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+export const init = createAsyncThunk(
+  'userPosts/fetchPosts',
+  (userId: number) => {
+    return getUserPosts(userId);
+  },
+);
+
+const initialState: PostsInfo = {
+  items: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const postsSlice = createSlice({
+  name: 'userPosts',
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder.addCase(init.pending, (state) => {
+      state.loaded = false;
+    });
+    builder.addCase(init.fulfilled, (state, action) => {
+      state.items = action.payload;
+      state.loaded = true;
+    });
+    builder.addCase(init.rejected, (state) => {
+      state.hasError = true;
```

Comment:
When an error occurs it might be helpful to store the error message to provide more detailed feedback to the user.

Reply:
This error is processed in App component and the user will see related markup. This logic was already in the task. Does it need to be changed?